### PR TITLE
Fix libspdm_rsa_generate_key always using the default key

### PR DESCRIPTION
**Pull request by Wenxing-hou**
Fix: #2336

Fix `libspdm_rsa_generate_key` always using the default key and add unit_test for `libspdm_rsa_generate_key`.

### CODE REVIEW

File: unit_test/test_crypt/rsa_verify.c [hunk 1]
```diff
@@ -103,6 +103,9 @@ uint8_t m_libspdm_rsa_pkcs1_signature[] = {
 /* Default public key 0x10001. */
 uint8_t m_libspdm_default_public_key[] = { 0x01, 0x00, 0x01 };
 
+/* input public key 0x03. */
+uint8_t m_libspdm_rsa_public_exponent[] = { 0x03 };
```
Comment (jyao1) [inline unit_test/test_crypt/rsa_verify.c:107]: rename to `m_libspdm_**test**_rsa_public_exponent`

Comment (Wenxing-hou) [reply unit_test/test_crypt/rsa_verify.c]: Thanks. I have fixed it.

File: unit_test/test_crypt/rsa_verify.c [hunk 2]
```diff
@@ -103,6 +103,19 @@ uint8_t m_libspdm_rsa_pkcs1_signature[] = {
 /* Default public key 0x10001. */
 uint8_t m_libspdm_default_public_key[] = { 0x01, 0x00, 0x01 };
 
+/* input public key 0x03. */
```
Comment (jyao1) [inline unit_test/test_crypt/rsa_verify.c:106]: why the comment is still 3?

File: unit_test/test_crypt/rsa_verify.c [hunk 3]
```diff
@@ -103,6 +103,19 @@ uint8_t m_libspdm_rsa_pkcs1_signature[] = {
...
+uint8_t m_libspdm_test_rsa_public_exponent1[] = { 0x03 };
+uint8_t m_libspdm_test_rsa_public_exponent2[] = { 0x02, 0x09 };
```
Comment (jyao1) [inline unit_test/test_crypt/rsa_verify.c:108]: why 0x92?

**Pull request merged**